.github/workflows/ci.yml

```diff
@@ -369,7 +369,7 @@ jobs:
 
     - name: Install core package
       run: |
-        pip install $(ls dist/*.whl)[all]
+        pip install $(ls dist/*.whl)
```

Comment:
Not sure if you want to have a full matrix of running against all installation flavours; IMO that would actually be a good idea...

Reply:
The idea here is to run all tests including the checklist ones. So, we install [all] at once.

Reply:
I understand, though that's kind of the root of not catching #5588 through CI before, because currently something like `allennlp test-install` is never tested without `checklist` being present. Hence my recommendation that this should likely be expanded to a full matrix of runs against each allennlp-flavour (base, -checklist and any other flavours that might be added), to ensure the CLI keeps working for all of those variants.

Reply:
Given that allennlp tests take a long time to run, I don't think we should run the full matrix for each flavor every time. @epwalsh Do you think it makes sense to do this when we make releases though?

Reply:
As a compromise, how about we run the full matrix in CI for the `test_package` job only? https://github.com/allenai/allennlp/blob/e1c6935cb6a95bb789a8a2a3d2fbbf4ab8239035/.github/workflows/ci.yml#L343

That should catch issues like #5588

Reply:
I did as suggested in https://github.com/allenai/allennlp/pull/5589/commits/e7d44a51f437c2f7d0abda82a44516da287c638e. :)